### ISSUE TITLE
'j0': The POSIX name for this item is deprecated. (VS2015) (Again)

### ISSUE THREAD

**Issue by KeithT** [ceres-solver/ceres-solver#281]
Just downloaded the latest master, built it on Windows 10, VS2015.

Then created a new project which uses jet.h and get the identical error to:
https://github.com/ceres-solver/ceres-solver/issues/190

**Comment by alexsmac** [comment]
Is CERES_MSVC_USE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS defined in the configured Ceres config.h (in <BUILD_DIR>/config/ceres/internal/config.h ?  And what appears in <BUILD_DIR>/CMakeFiles/CMakeOutput.log related to: HAVE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS ?

**Comment by KeithT** [comment]
Thanks for the support.

[CMakeOutput.txt](https://github.com/ceres-solver/ceres-solver/files/988765/CMakeOutput.txt)

Yes I believe so, from config.h:
"// If defined, Ceres was compiled with a version MSVC >= 2005 which
// deprecated the standard POSIX names for bessel functions, replacing them
// with underscore prefixed versions (e.g. j0() -> _j0()).
#define CERES_MSVC_USE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS"

and this is the line from CMakeOutput.Log (complete file attached).
"Performing C++ SOURCE FILE Test HAVE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS succeeded with the following output:
Change Dir: C:/install/ceres/ceres-solver-master/ProjectSharedLib/CMakeFiles/CMakeTmp"

**Comment by KeithT** [comment]
Further to the above, when I open jet.h in the ceres project the #if defined (CERES_MSVC_USE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS) condition is active, whereas in the other project that links to ceres the else condition is. I guess this is because there are no "Preprocessor Definitions" set in the c/c++ propoerties in the new project, so CERES_MSVC_USE_UNDERSCORE_PREFIXED_BESSEL_FUNCTIONS is not defined. I added the directive in a new config.h file in the new project and it solved the issue, but this obviously creates a dependency to do this for any projects using ceres.

**Comment by alexsmac** [comment]
This sounds like you have a configuration issue and you are including the headers for a different version of Ceres in your project.

jet.h #includes ceres/internal/port.h, which in turn #includes ceres/internal/config.h provided that you are compiling for C++ (which I'm assuming you are given that you are trying to use jet.h).

So whilst jet.h should really #include ceres/internal/config.h directly, doing so would not change the behaviour from what it is now when it should already be being included.

Do you have an installed version of Ceres whose headers are on the #include path for your external project, perhaps due to another package?  That was a failure mode for people on *nix when they had a version of Ceres installed in /usr/local (as are its public dependencies, e.g. glog), but are trying to build/use another version of Ceres located elsewhere, as ceres/internal/config.h can be found via the installed /usr/local version which must be on the path for the public dependencies to be found.

**Comment by KeithT** [comment]
There were 2 versions installed, although I am pretty sure only 1 was referenced in the project. I started again to be sure.

What I am trying to do is build ceres and have a new project (outside the ceres VS solution) link to it. (c++, Windows 10, VS2015).

I have eigen, glog, gflags downloaded and built were necessary and have 2 ceres configs one with examples and one without, also built.

As a first step I am trying to get the helloworld example working in a new project that links to the "no examples" ceres build. I tried to do this manually setting up include dirs and links and got it down to 4 link errors, similar to the ones in this issue: https://github.com/ceres-solver/ceres-solver/issues/227. I'm now trying to create a cmake file to do this. The link about this http://ceres-solver.org/building.html#using-ceres-with-cmake referenced in issue 227 no longer works. Is this available anywhere else?

Thanks again for any help.
